### PR TITLE
feat: iOS replay support

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description

adds iOS to the supported platforms for replay.

## :bulb: Motivation and Context

part of #1193 

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/ios/Classes/SentryFlutterReplayScreenshotProvider.m [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+- (instancetype _Nonnull)initWithChannel:
+    (FlutterMethodChannel *_Nonnull)channel {
+  if (self = [super init]) {
+    self->channel = channel;
+  }
+  return self;
+}
+
```
Comment (buenaflor) [inline flutter/ios/Classes/SentryFlutterReplayScreenshotProvider.m:18]: curious why we need to implement screenshot provider and breadcrumb converter in objc, is it for performance?

File: flutter/lib/src/native/cocoa/sentry_native_cocoa.dart [hunk 2]
```diff
@@ -1,17 +1,76 @@
...
+              _replayId = replayId;
+              hub.configureScope((s) {
+                // ignore: invalid_use_of_internal_member
+                s.replayId = replayId;
+              });
+            }
+
+            Uint8List? imageBytes;
+            await _replayRecorder?.capture((image) async {
+              var imageData =
```
Comment (buenaflor) [inline flutter/lib/src/native/cocoa/sentry_native_cocoa.dart]:
```suggestion
              final imageData =
```

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 3]
```diff
@@ -174,6 +179,14 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
         case "resumeAppHangTracking":
             resumeAppHangTracking(result)
 
+        case "sendReplayForEvent":
+#if canImport(UIKit) && !SENTRY_NO_UIKIT && (os(iOS) || os(tvOS))
```
Comment (buenaflor) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift:183]: `|| os(tvOS)` 

we also support replay for tvos?

**Review by buenaflor — APPROVED** [review]
looks good, just a couple minor comments/questions

File: flutter/ios/Classes/SentryFlutterPluginApple.swift (hunk 3 above)
Comment (vaind) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift:183]: no (it's not even supported by Flutter officially AFAIK), the condition is exactly the same as in cocoa core code

File: flutter/ios/Classes/SentryFlutterReplayScreenshotProvider.m (hunk 1 above)
Comment (vaind) [reply flutter/ios/Classes/SentryFlutterReplayScreenshotProvider.m:18]: the types we need from cocoa are not visible outside their module so we need to do it in objc which ignores swift visibility rules.

**Pull request merged**